lib/ansible/module_utils/basic.py

```diff
@@ -2607,9 +2602,15 @@ def run_command(self, args, check_rc=False, close_fds=True, executable=None, dat
                 cmd.stdin.close()
 
             while True:
-                rfds, wfds, efds = select.select(rpipes, [], rpipes, 1)
-                stdout += self._read_from_pipes(rpipes, rfds, cmd.stdout)
-                stderr += self._read_from_pipes(rpipes, rfds, cmd.stderr)
+                events = selector.select(1)
+                for key, event in events:
+                    b_chunk = os.read(key.fileobj.fileno(), self.get_buffer_size(key.fileobj))
```

Comment:
* Is there a reason for using `os.read()` instead of `key.fileobj.read()`?
* Is there a reason for using `get_buffer_size()` rather than just letting `read()` slurp in all of the data that's available?

Reply:
I got some failures with tests and I'm not actually that well versed with all the idioms around testing and mocking … thus I just preserved the currently present impl.

Reply:
Okay... We shouldn't let tests drive implementation (unless it's a case where the implementation is more modular, easier to read, or more flexible once it's adapted to the test case).  let me take a look at updating the test case.

Reply:
I pushed this change in my commit to the code